File: java/lib/SnobotSim/ModuleWrapper/SolenoidWrapper.cpp
/*
 * SolenoidWrapper.cpp
 *
 *  Created on: May 3, 2017
 *      Author: PJ
 */

#include "SnobotSim/ModuleWrapper/SolenoidWrapper.h"

SolenoidWrapper::SolenoidWrapper(int aPort) :
        AModuleWrapper("Solenoid " + std::to_string(aPort)), mState(false)
{

}

SolenoidWrapper::~SolenoidWrapper()
{

}


void SolenoidWrapper::SetState(bool aOn)
{
    mState = aOn;
}

bool SolenoidWrapper::GetState()
{
    return mState;
}

File: snobot_sim_java/src/main/java/com/snobot/simulator/module_wrapper/SolenoidWrapper.java
package com.snobot.simulator.module_wrapper;

public class SolenoidWrapper extends ASensorWrapper
{
    private boolean mState;

    public SolenoidWrapper(int aIndex)
    {
        super("Solenoid " + aIndex);
        mState = false;
    }

    public boolean get()
    {
        return mState;
    }

    public void set(boolean aState)
    {
        mState = aState;
    }
}
